tests/test_audit.py

```diff
@@ -10,23 +13,157 @@
+        # Single exclude actor
+        (None, None, ["exclude-user"], None),
+        # Multiple exclude actors
+        (None, None, ["exclude-user1", "exclude-user2"], None),
+        # Actions + actors
+        (["org.create", "repo.destroy"], ["octocat"], None, None),
+        # Actions + exclude_actors
+        (["team.add_member"], None, ["bot-user"], None),
+        # Actors + exclude_actors
+        (None, ["octocat", "monalisa"], ["bot-user"], None),
+        # All parameters combined
+        (
+            ["protected_branch.create", "repo.download_zip"],
+            ["octocat"],
+            ["exclude-user"],
+            None,
+        ),
+        # Maximum complexity
+        (
+            ["org.create", "team.add_member", "repo.destroy"],
+            ["octocat", "monalisa", "admin-user"],
+            ["bot-user1", "bot-user2", "exclude-admin"],
+            None,
+        ),
+        # All None values
+        (None, None, None, None),
+        # Empty lists
+        ([], [], [], None),
+    ],
+)
+@pytest.mark.asyncio
+async def test_get_audit_parameterized(
+    gh_rest_mock: GithubHttpxMock,
+    audit_extractor: Callable[
+        [list[str] | None, list[str] | None, list[str] | None, dict[str, int] | None],
+        GithubAuditLogExtractor,
+    ],
+    actions: list[str] | None,
+    actors: list[str] | None,
+    exclude_actors: list[str] | None,
+    lookback_period: dict[str, int] | None,
+):
+    extractor = audit_extractor(
+        actions=actions,
+        actors=actors,
+        exclude_actors=exclude_actors,
+        lookback_period=lookback_period,
     )
 
+    expected_search_phrase = build_search_phrase(
+        actions=actions or [],
+        actors=actors or [],
+        exclude_actors=exclude_actors or [],
+        lookback_period={},
+    )
 
+    if expected_search_phrase:
+        gh_rest_mock.get_enterprise_audit_logs(
+            status_code=200,
+            search_phrase=expected_search_phrase,
+            json=GITHUB_AUDIT,
+        )
+    else:
+        gh_rest_mock.add_response(
+            url=f"{gh_rest_mock.base_url}/enterprises/test-enterprise/audit-log?per_page=100",
+            status_code=200,
+            json=GITHUB_AUDIT,
+        )
+
+    all_records = [record async for record in extractor.extract_records()]
+    assert all_records == GITHUB_EXPECTED_OUTPUT
+
+
+@pytest.mark.parametrize(
+    "lookback_period",
+    [
+        {"days": 7},
+        {"months": 2},
+        {"years": 1},
+        {"days": 15, "months": 1},
+        {"days": 10, "months": 1, "years": 1},
+    ],
+)
 @pytest.mark.asyncio
-async def test_get_audit(
-    audit_extractor: GithubAuditLogExtractor, gh_rest_mock: GithubHttpxMock
+async def test_get_audit_lookback_periods(
+    gh_rest_mock: GithubHttpxMock,
+    audit_extractor: Callable[
+        [list[str] | None, list[str] | None, list[str] | None, dict[str, int] | None],
+        GithubAuditLogExtractor,
+    ],
```

Comment:
Since this is only used in three tests, it's ok to initialize the extractor anew in each test.